Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,250 @@
+                        extensions = vcamExtensions.ToArray(),
+                        custom_extension_count = customExtensionCount,
+                    });
+
+                    var vcamChildren = vcamBase.GetComponentsInChildren<CinemachineVirtualCamera>();
+                    for (var c = 0; c < vcamChildren.Length; c++)
+                    {
+                        vcamDatas.Add(ConvertVcamToVcamData(vcamChildren[c], id + "." + c));
+                    }
+                }
+            }
+
+            var projectData = new ProjectData
+            {
+                brain_count = cinemachineCore.BrainCount,
+                vcam_count = cinemachineCore.VirtualCameraCount,
+                cam_count = Camera.allCamerasCount,
+                vcams = vcamDatas,
+                time_elapsed = Time.realtimeSinceStartup - startTime,
+            };
+
+            // Register our event like this
+            EditorAnalytics.RegisterEventWithLimit("cm_vcams_on_play", k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+
+            // Send the data to the database
+            EditorAnalytics.SendEventWithLimit("cm_vcams_on_play", projectData);
+        }
+
+        [Serializable]
+        struct ProjectData
+        {
+            public int brain_count;
+            public int vcam_count;
+            public int cam_count;
+            public List<VcamData> vcams;
+            public float time_elapsed;
+        }
+
+        [Serializable]
+        struct VcamData
+        {
+            public string id;
+            public string vcam_class;
+            public bool has_follow_target;
+            public bool has_lookat_target;
+            public string blend_hint;
+            public bool inherit_position;
+            public string standby_update;
+            public string mode_overwrite;
+            public string body_component;
+            public string aim_component;
+            public string noise_component;
+            public int custom_component_count;
+            public string[] extensions;
+            public int custom_extension_count;
+        }
+
+        static readonly VcamData k_nullData = new VcamData { vcam_class = "null" };
+        const string k_customStr = "Custom"; // used for hiding user created class names
+
+        static VcamData ConvertVcamToVcamData(in CinemachineVirtualCamera vcam, string id)
+        {
+            if (vcam == null) return k_nullData;
+
+            // collect extensions
+            GetExtensions(vcam, out List<string> vcamExtensions, out int customExtensionCount);
+
+            // collect components on vcam
+            int customComponentCount = 0;
+            string bodyComponent = "", aimComponent = "", noiseComponent = "";
+            var cmComps = vcam.GetComponentPipeline();
+            if (cmComps != null)
+            {
+                foreach (var cmComp in cmComps)
+                {
+                    var type = cmComp.GetType();
```

Comment:
ok, but low marks for style.  How much more can you simplify this file if you introduce this method?

        static string GetTypeName(Type type, ref int customTypeCount)
        {
            if (typeof(CinemachineBrain).Assembly != type.Assembly)
            {
                ++customTypeCount;
                return k_customStr;
            }
            return type.ToString().Substring(12); // 12 = "Cinemachine.".Length
        }

Reply:
...and what about using type.Name instead of type.ToString().Substring(12)

Reply:
typeof(T).Name // class name, no namespace
typeof(T).FullName // namespace and class name
typeof(T).Namespace // namespace, no class name